spec/Spec/KZG.cry

```diff
@@ -72,3 +74,27 @@ compute_kzg_proof blob z_bytes = (proof, y) where
     polynomial = blob_to_polynomial blob
     (proof, y') = compute_kzg_proof_impl polynomial (bytes_to_bls_field z_bytes)
     y = split`{BYTES_PER_FIELD_ELEMENT} y'
+
+/**
+ * Verify KZG proof that `p(z) == y` where `p(z)` is the polynomial represented by `polynomial_kzg`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#verify_kzg_proof_impl
+ */
+verify_kzg_proof_impl : KZGCommitment -> BLSFieldElement -> BLSFieldElement -> KZGProof -> Bit
+verify_kzg_proof_impl commitment z y proof = pairing_check points where
+    // Verify: P - y = Q * (X - z)
+    x_minus_z = g2_add (bytes96_to_G2 (KZG_SETUP_G2_MONOMIAL@1)) g2_mult
+    g2_mult = g2_multi_optimized G2 ((BLS_MODULUS - z) % BLS_MODULUS)
+    p_minus_y = g1_add (bytes48_to_G1 commitment) g1_mult
+    g1_mult = g1_multi_optimized G1 ((BLS_MODULUS - y) % BLS_MODULUS)
+    points = [(p_minus_y, (g2_neg G2)), ((bytes48_to_G1 proof), x_minus_z)]
+
+/**
+ * Verify KZG proof that `p(z) == y` where `p(z)` is the polynomial represented by `polynomial_kzg`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#verify_kzg_proof
```

Comment:
We are planning on adding some tests for this correct?

Reply:
Yeah, that's what I'm trying to run right now. Maybe I should have made this a draft PR ... since I want to include some testing first.